src/main/kotlin/org/rust/ide/actions/RsJoinRawLinesHandler.kt

```diff
@@ -19,9 +18,23 @@ class RsJoinRawLinesHandler : JoinRawLinesHandlerDelegate {
     override fun tryJoinRawLines(document: Document, file: PsiFile, start: Int, end: Int): Int {
         if (file !is RsFile) return CANNOT_JOIN
         if (start == 0) return CANNOT_JOIN
+
+        val joinStruct = tryJoinStruct(file, start)
+        if (joinStruct != CANNOT_JOIN)
+            return joinStruct
+
         return tryJoinSingleExpressionBlock(file, start)
     }
 
+    fun tryJoinStruct(file: RsFile, start: Int): Int {
+        val body = file.findElementAt(start)!!.parent
```

Comment:
Hm, I think we need a somewhat more precise check here. Perhaps using `.parentOfType<RsStructLiteral>` and 
checking that the opening brace and the caret are on the same lines. otherwise this example won't work

```
let _ = S {
   foo: 92 <caret>,
}
```

And that gets me thinking about what actual behavior do we want here? Joining the *whole* structure to the single seems a bit out of scope for joining lines (that means that my original test case is wrong). The better behavior would be to just join two lines, as usual, but to make sure that if `}` is on the same line as the last field, there's no trailing comma. It would be nice to check if it works well with selection as well!

So, I am thinking about the following test cases:

```rust
S { |
    foo: 92,
}

->

S { foo: 92, |
}

->

S { foo: 92 }

---------------------------------

S {
  foo: 92,
  bar: |92,
}

->

S {
  foo: 92,
  bar: 92 }

---------------------------------

S <selection>{
  foo: 92,
  bar: 92,
}</selection>

->

S { foo: 92, bar: 92 }
```